packages/parser-javascript/src/parser.ts

```diff
@@ -29,7 +29,8 @@ export default class JavascriptParser extends Parser {
 
         const scriptData: ScriptParse = {
             resource,
-            sourceCode: new eslint.SourceCode(code, ast)
+            sourceCode: new eslint.SourceCode(code, ast),
+            tokenCount: ast && ast.tokens && ast.tokens.length || 0
```

Comment:
I initially considered doing that. But I felt that, passing the entire ast is unnecessary **at this point of time**.  I am not an expert in node/js memory allocation/call stacks. But a simple value type value is definitely smaller memory foot print than a bigger object.

If another rule requires it in future, we can definitely pass the ast at that time.  I am more of a "**use it when you absolutely need it**" mindset person. 

If you still strongly feel we should pass the tree, we can make the change. Let me know.

Reply:
The `ast` is already calculated and used inside `eslint.SourceCode` so it's not like we are going to use less memory. I think we should pass it or otherwise it will be a breaking change when/if we add it (and  I have a feeling we might need it in the near future).

Thanks!

Reply:
All right. Will do.